src/ast/irbuilderbpf.cpp

```diff
@@ -473,15 +473,27 @@ CallInst *IRBuilderBPF::CreateGetStrScratchMap(int idx,
                              idx);
 }
 
-CallInst *IRBuilderBPF::CreateGetFmtStringArgsScratchMap(
-    BasicBlock *failure_callback,
-    const location &loc)
+Value *IRBuilderBPF::CreateGetFmtStringArgsScratchBuffer(const location &loc)
 {
-  return createGetScratchMap(to_string(MapType::FmtStringArgs),
-                             "fmtstr",
-                             GET_PTR_TY(),
-                             loc,
-                             failure_callback);
+  auto config = globalvars::get_config(
+      bpftrace::globalvars::GlobalVar::FMT_STRINGS_BUFFER, bpftrace_.resources);
+
+  // Get CPU ID
+  auto cpu_id = CreateGetCpuId(loc);
+  auto max = CreateLoad(getInt64Ty(),
+                        module_.getGlobalVariable(to_string(
+                            bpftrace::globalvars::GlobalVar::MAX_CPU_ID)));
+  auto cmp = CreateICmp(CmpInst::ICMP_ULE, cpu_id, max, "cpuid.min.cmp");
+  auto select = CreateSelect(cmp, cpu_id, max, "cpuid.min.select");
```

Comment:
Maybe add a small comment about why this comparison is necessary

Reply:
Fixed in latest PR update (coming soon)